pkg/monitor/datapath_trace.go

```diff
@@ -77,6 +94,36 @@ func connState(reason uint8) string {
 	return fmt.Sprintf("%d", reason)
 }
 
+// DecodeTraceNotify will decode 'data' into the provided TraceNotify structure
+func DecodeTraceNotify(data []byte, tn *TraceNotify) error {
+	if len(data) < traceNotifyCommonLen {
+		return fmt.Errorf("Unknown trace event")
+	}
+
+	var msgHdrLen uint16
+	// metaLenOffset := unsafe.Offsetof(TraceNotify.MetaLen)
+	metaLenOffset := 14
+	// metaLenSize := unsafe.Sizeof(TraceNotify.MetaLen)
+	metaLenSize := 2
+	metaLen := data[metaLenOffset : metaLenOffset+metaLenSize]
+	err := binary.Read(bytes.NewReader(metaLen), byteorder.Native, &msgHdrLen)
+	if err != nil {
+		return err
+	}
+
+	switch msgHdrLen {
```

Comment:
Got a bit confused here. So this basically uses the msgHdrLen as a version type for the decoder? Any reason why not to call it `msgHdrVersion` and just enum that field with the supported types? Otherwise in the future if we ever add another message type that happens to have the same size as V2Len it will be thrown off.